src/modules/interfaces.ts

```diff
@@ -782,6 +797,62 @@ export class ModuleBase {
     return [];
   }
 
+  getFunctionsFromSql(sql: string): {
+    [key: string]: { description?: string; sample_usage?: string; signature?: string };
+  } {
+    const returnValue: {
+      [key: string]: { description?: string; sample_usage?: string; signature?: string };
+    } = {};
+    try {
+      const sqlParsed = parse(sql);
+      const functionsAndParams = sqlParsed
+        .filter((s: any) => _.has(s, ['RawStmt', 'stmt', 'CreateFunctionStmt']))
+        .map((s: any) => s.RawStmt.stmt.CreateFunctionStmt)
+        .map((s: any) => [s.funcname?.pop()?.String?.str, s.parameters])
+        .filter((v: any[]) => !!v[0]);
+
+      for (const [functionName, functionParams] of functionsAndParams) {
+        const inputs: string[] = [],
+          outputs: string[] = [];
+        functionParams?.map((p: any) => {
+          if (['FUNC_PARAM_IN', 'FUNC_PARAM_VARIADIC'].includes(p.FunctionParameter.mode)) {
+            const variadic = p.FunctionParameter.mode === 'FUNC_PARAM_VARIADIC';
+            inputs.push(
+              `${variadic ? 'variadic ' : ''}${p.FunctionParameter.name} ${
+                p.FunctionParameter.argType?.names?.pop()?.String?.str
+              }${variadic ? '[]' : ''}`,
+            );
+          } else
+            outputs.push(
+              `${p.FunctionParameter.name} ${p.FunctionParameter.argType?.names?.pop()?.String?.str}`,
+            );
+        });
+        returnValue[functionName] = { signature: `(${inputs.join(', ')}) => (${outputs.join(', ')})` };
+      }
+
+      sqlParsed
+        .filter((s: any) => _.has(s, ['RawStmt', 'stmt', 'CommentStmt']))
+        .map((s: any) => {
+          return [
+            s.RawStmt.stmt.CommentStmt.object?.ObjectWithArgs?.objname?.pop().String?.str,
+            safeParse(s.RawStmt.stmt.CommentStmt.comment),
+          ];
+        })
+        .filter((v: any[]) => !!v[0] && !!v[1])
+        .map((v: [string, { description?: string; sample_usage?: string }]) => {
+          returnValue[v[0]] = {
+            ...returnValue[v[0]],
+            description: v[1].description,
+            sample_usage: v[1].sample_usage,
+          };
+        });
+      return returnValue;
+    } catch (_) {
+      /** Do nothing */
+    }
+    return {};
```

Comment:
well this could be
```suggestion
    return returnValue;
```
right? and the return in line 849 should not be necessary?

Reply:
In the current version it either returns a correctly created object or an empty one. In that case it may return partially-filed objects. Which one do you think we should do?

Reply:
let's keep it as is